src/SFML/Window/Android/WindowImplAndroid.cpp

```diff
@@ -461,6 +458,20 @@ int WindowImplAndroid::processKeyEvent(AInputEvent* inputEvent, ActivityStates&
     return 0;
 }
 
+////////////////////////////////////////////////////////////
+int WindowImplAndroid::processJoystickButtonEvent(AInputEvent*     inputEvent,
+                                                  std::int32_t     action,
+                                                  Joystick::Button button,
+                                                  ActivityStates&  states)
+{
+    const auto deviceId = AInputEvent_getDeviceId(inputEvent);
+    if (states.joystickStates.find(deviceId) == states.joystickStates.end())
+        return 1;
+
+    const auto buttonIdx = static_cast<std::underlying_type_t<decltype(button)>>(button);
+    states.joystickStates.at(deviceId).buttons[buttonIdx] = action == AKEY_EVENT_ACTION_DOWN;
```

Comment:
Direct access should be fine here, as we just checked it above:

```suggestion
    states.joystickStates[deviceId].buttons[buttonIdx] = action == AKEY_EVENT_ACTION_DOWN;
```